scapy/contrib/pnio_dcp.py

```diff
@@ -0,0 +1,559 @@
+    },
+    # device properties
+    0x02: {
+        0x00: "Reserved",
+        0x01: "Manufacturer specific (Type of Station)",
+        0x02: "Name of Station",
+        0x03: "Device ID",
+        0x04: "Device Role",
+        0x05: "Device Options",
+        0x06: "Alias Name",
+        0x07: "Device Instance",
+        0x08: "OEM Device ID",
+    },
+    # dhcp
+    0x03: {
+        0x0c: "Host name",
+        0x2b: "Vendor specific",
+        0x36: "Server identifier",
+        0x37: "Parameter request list",
+        0x3c: "Class identifier",
+        0x3d: "DHCP client identifier",
+        0x51: "FQDN, Fully Qualified Domain Name",
+        0x61: "UUID/GUID-based Client",
+        0xff: "Control DHCP for address resolution"
+    },
+    # control
+    0x05: {
+        0x00: "Reserved",
+        0x01: "Start Transaction",
+        0x02: "End Transaction",
+        0x03: "Signal",
+        0x04: "Response",
+        0x05: "Reset Factory Settings",
+        0x06: "Reset to Factory"
+    },
+    # device initiative
+    0x06: {
+        0: "Reserved",
+        1: "Device Initiative"
+    },
+    0xff:
+        {
+            0xff: "ALL Selector"
+        }
+}
+
+BLOCK_INFOS = {
+    0x00: "Reserved",
+    range(0x01, 0xff): "Reserved",
+}
+
+IP_BLOCK_INFOS = {
+    0x0000: "IP not set",
+    0x0001: "IP set",
+    0x0002: "IP set by DHCP",
+    range(0x0003, 0x007f): "reserved",
+    0x0080: "IP not set (address conflict detected)",
+    0x0081: "IP set (address conflict detected)",
+    0x0082: "IP set by DHCP (address conflict detected)",
+}
+
+BLOCK_ERRORS = {
+    0x00: "Ok",
+    0x01: "Option unsupp.",
+    0x02: "Suboption unsupp. or no DataSet avail.",
+    0x03: "Suboption not set",
+    0x04: "Resource Error",
+    0x05: "SET not possible by local reasons",
+    0x06: "In operation, SET not possible",
+}
+
+BLOCK_QUALIFIERS = {
+    0x0000: "Use the value temporary",
+    0x0001: "Save the value permanent",
+    range(0x0002, 0x00ff): "reserved"
+}
+
+
+#####################################################
+#                     DCP Blocks                    #
+#####################################################
+
+# GENERIC DCP BLOCK
+# similar problem in https://stackoverflow.com/questions/33756348/scapy-using-a-packetlistfield-to-dissect-multiple-packets-contained-in-a-packet
+
+# DCP RESPONSE BLOCKS
+
+class DCPBaseBlock(Packet):
+    """
+        base class for all DCP Blocks
+    """
+    fields_desc = [
+        ByteEnumField("option", 1, DCP_OPTIONS),
+        MultiEnumField("sub_option", 2, DCP_SUBOPTIONS, fmt='B', depends_on=lambda p: p.option),
+        FieldLenField("dcp_block_length", None, length_of="data"),  # TODO
+        ShortEnumField("block_info", 0, BLOCK_INFOS),
+        StrLenField("data", "", length_from=lambda x: x.dcp_block_length),
+    ]
+
+    # needed for dissecting multiple blocks
+    # https://stackoverflow.com/questions/8073508/scapy-adding-new-protocol-with-complex-field-groupings
```

Comment:
That's not an issue, therefore there is no need to link the question :smile:

That's something we should document more clearly